src/main/java/org/example/homework03/domain/repository/UserRepository.java

```diff
@@ -0,0 +1,13 @@
+package org.example.homework03.domain.repository;
+
+import org.example.homework03.entity.User;
+
+import java.util.Optional;
+
+public interface UserRepository {
+    User persistUser(User user);
+
+    Optional<User> getUserById(Long id);
+
+    Optional<User> getUserByEmail(String email);
+}
```

Comment:
Optional comment: You already calling method from userRepository, so you can simplify method names.

Also, if we want to return Optional, it is better to start method name with `find`. Is you will return the object immediately (not Optional) than you can use `get...` method name
```suggestion
public interface UserRepository {
    User persist(User user);

    Optional<User> findById(Long id);

    Optional<User> findByEmail(String email);
}
```